src/tagmanager/tm_workspace.c

```diff
@@ -742,9 +797,54 @@ static void fill_find_tags_array_prefix(GPtrArray *dst, const char *name,
 }
 
 
+/* fill 'includes' with the included files except the header corresponding
+ * to the current file which is returned in 'header' */
+static void fill_includes(TMSourceFile *source, TMSourceFile **header, GHashTable *includes)
```

Comment:
> I would find it idiomatic to make this get_includes() that returns the included files list.

This is a relict of when I experimented with running this function recursively. Since it didn't work very well, it indeed looks better as you describe.

> Also, I find it strange that the corresponding header is excluded from the included files list. I think it wouldn't hurt because sort_found_tags() is special-casing the corresponding header anyway and might make this function be useful in other situations as well.

Can do.